src/fu-util.c

```diff
@@ -894,6 +894,20 @@ fu_util_device_test_step(FuUtilPrivate *priv,
 						 priv->cancellable,
 						 error))
 			return FALSE;
+
+		/* clear json files from cache to avoid back to back test failures */
+		if (g_str_has_suffix(emulation_filename, ".json")) {
+			g_autofree gchar *cache = fu_util_get_user_cache_path(emulation_url);
+
+			if (g_file_test(cache, G_FILE_TEST_EXISTS)) {
+				g_autoptr(GFile) file = g_file_new_for_path(cache);
+
+				if (!g_file_delete(file, NULL, error)) {
```

Comment:
Not sure about this; the idea of the cache is that it means we don't need to download it next time. I think saving the file with a SHA prefix is probably a good plan.

Reply:
By moving in repo, completely sidestep the caching problem.